src/dataServices/dynamoDbBundleServiceHelper.ts

```diff
@@ -253,4 +256,211 @@ export default class DynamoDbBundleServiceHelper {
+                        resource: {},
+                        resourceType: request.resourceType,
+                    });
+                    break;
+                }
+                case 'read': {
+                    if (vid === 0) {
+                        break;
+                    }
+                    batchReadWriteResponses.push({
+                        id,
+                        vid: vid.toString(),
+                        operation: request.operation,
+                        lastModified: '',
+                        resource: item?.resource,
+                        resourceType: request.resourceType,
+                    });
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+        // we cannot do deleteRequests nor updateRequests in a batchwriteitem call, since we use the update api instead of delete
+        // hence, we will separate these requests and use batchexecuteStatement to update items in a batch (and
+        // we know there are no conflicts since there will only be updates running)
+        return {
+            editRequests: [...deleteRequests, ...createRequests, ...updateRequests],
+            batchReadWriteResponses,
+        };
+    }
+
+    static async processBatchEditRequests(
+        editRequests: any[],
+        batchReadWriteResponses: BatchReadWriteResponse[],
+        dynamoDb: DynamoDB,
+    ) {
+        const updatedResponses = batchReadWriteResponses;
+        for (let i = 0; i < editRequests.length; i += MAX_BATCH_WRITE_ITEMS) {
+            const upperLimit = Math.min(i + MAX_BATCH_WRITE_ITEMS, editRequests.length);
+            const batch = editRequests.slice(i, upperLimit);
+            const statements = batch.map((x) => {
+                return { Statement: x.Statement };
+            });
+            // eslint-disable-next-line no-await-in-loop
+            const batchExecuteResponse = await dynamoDb
+                .batchExecuteStatement({
+                    Statements: [...statements],
+                })
+                .promise();
+            batchExecuteResponse?.Responses?.forEach((response, index) => {
+                if (response.Error) {
+                    console.log('Unable to process request: ', response.Error);
+                    const indexOfFailure = batchReadWriteResponses.findIndex((x) => x.id === batch[index].id);
+                    if (index !== -1) {
+                        updatedResponses[indexOfFailure] = {
+                            ...batchReadWriteResponses[indexOfFailure],
+                            vid: '',
+                        };
+                    }
+                }
+            });
+        }
+        return updatedResponses;
+    }
+
+    // ExecuteStatement requires objects to be in the form a PartiQL tuple
+    // this means strings must be single quotes
+    static convertToPartiQLString(value: any) {
```

Comment:
This looks hard to read and maintain, is there a library or something that does this? If using this function is unavoidable for inserting with batchExecuteStatement method, maybe we should switch back to the batch writes item method, which does not require parsing the object like this.

Reply:
I did search for a library to do this, but couldn't find anything through the AWS docs or any other libraries. I will switch back to batchWriteItem for both creates and updates 👍